### PR TITLE
change WebValve.register to accept a class name instead of class

### PR DESCRIPTION
**Pull request by samandmoore**
rails 6 autoloading emits a deprecation warning for class loading during
app initialization. webvalve's register call is a culprit here, but it
doesn't have to be. this changeset makes it so that webvalve accepts a
stringified class name instead of a class and makes that all work
properly with the internals.

/domain @smudge @jmileham @effron @aburgel 
/no-platform

more info on this change can be found here https://github.com/rails/rails/issues/36363
the tl;dr for us is that we could remove webvalve things from the autoload paths, but that would make local dev more tedious. so, instead, we just don't load the class until after initialization.

### CODE REVIEW

File: examples/sinatra/webvalve/fake_twitter.rb [hunk 1]
```diff
@@ -1,6 +1,4 @@
 class FakeTwitter < WebValve::FakeService
-  URL = 'http://faketwitter.test'.freeze
```
Comment (samandmoore) [inline examples/sinatra/webvalve/fake_twitter.rb]: we can't do this sorta thing anymore. we'd have to put this URL into lib or some other non autoloaded path. i was never a huge fan of this approach, so i'm not worried about it.

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,9 +9,13 @@ and this project aims to adhere to [Semantic Versioning](http://semver.org/spec/
 ### Added
 ### Removed
 
+## [0.9.11] - 2019-09-23
```
Comment (smudge) [inline CHANGELOG.md]: This should go in version.rb too, right?

File: .travis.yml [hunk 3]
```diff
@@ -2,6 +2,7 @@ language: ruby
 rvm:
   - 2.3.1
   - 2.4.2
+  - 2.5.1
```
Comment (smudge) [inline .travis.yml]: Should 2.6.1 go in this list?

**Comment by smudge** [comment]
<< domain LGTM :+1:

File: CHANGELOG.md (hunk 2 above)
Comment (samandmoore) [reply CHANGELOG.md]: Yea I'm gonna update before merge.

File: .travis.yml (hunk 3 above)
Comment (samandmoore) [reply .travis.yml]: Yea might as well. I gotta fix the rails 6 thing to make sure we exclude < 2.5 in ci

File: lib/webvalve/fake_service_config.rb [hunk 4]
```diff
@@ -26,9 +26,9 @@ def service_url
 
     def missing_url_message
       <<~MESSAGE
-        There is no URL defined for #{service.name}.
+        There is no URL defined for #{service_class_name}.
         Configure one by setting the ENV variable "#{service_name.to_s.upcase}_API_URL"
-        or by using WebValve.register #{service.name}, url: "http://something.dev"
+        or by using WebValve.register "#{service_class_name}", url: "http://something.dev"
```
Comment (phantomphildius) [inline lib/webvalve/fake_service_config.rb]: im a little late to the party with this nit but should this message have the evaluated service_class_name in quotes so no one gets confused and uses the now deprecated way of registering a service?

Comment (phantomphildius) [reply lib/webvalve/fake_service_config.rb]: I guess that's handled globally in the manager class

Comment (samandmoore) [reply lib/webvalve/fake_service_config.rb]: it's in quotes here, no? since this string is inside a heredoc the quotes are literals
